src/java/org/apache/cassandra/tools/nodetool/CompactionStats.java

```diff
@@ -53,29 +60,81 @@ public class CompactionStats extends NodeToolCmd
             for (Entry<String, Integer> tableEntry : ksEntry.getValue().entrySet())
-                numTotalPendingTask += tableEntry.getValue();
+            {
+                toPrint.append("- " + ksName + '.' + tableEntry.getKey() + ": " + tableEntry.getValue());
+                toPrint.append(LINE_SEPARATOR);
+            }
         }
-        out.println("pending tasks: " + numTotalPendingTask);
+
+        return toPrint.toString();
+    }
+
+    private static int numPendingTasks(Map<String, Map<String, Integer>> pendingTaskNumberByTable)
+    {
+        int numTotalPendingTasks = 0;
         for (Entry<String, Map<String, Integer>> ksEntry : pendingTaskNumberByTable.entrySet())
         {
-            String ksName = ksEntry.getKey();
             for (Entry<String, Integer> tableEntry : ksEntry.getValue().entrySet())
-            {
-                String tableName = tableEntry.getKey();
-                int pendingTaskCount = tableEntry.getValue();
-
-                out.println("- " + ksName + '.' + tableName + ": " + pendingTaskCount);
-            }
+                numTotalPendingTasks += tableEntry.getValue();
         }
-        out.println();
-        reportCompactionTable(cm.getCompactions(), probe.getCompactionThroughputBytes(), humanReadable, vtableOutput, out);
+
+        return numTotalPendingTasks;
+    }
+
+    private static int numConcurrentCompactorsConfigured(NodeProbe probe)
+    {
+        return probe.getConcurrentCompactors();
+    }
+
+    private static String compactionsCompletedStats(NodeProbe probe)
+    {
+        Long completedTasks = (Long)probe.getCompactionMetric("CompletedTasks");
+        CassandraMetricsRegistry.JmxMeterMBean totalCompactionsCompletedMetrics =
+            (CassandraMetricsRegistry.JmxMeterMBean)probe.getCompactionMetric("TotalCompactionsCompleted");
+        NumberFormat formatter = new DecimalFormat("##.00");
+        StringBuffer toPrint = new StringBuffer();
+        toPrint.append(String.format("compactions completed: %s", completedTasks));
+        toPrint.append(LINE_SEPARATOR);
+        toPrint.append(String.format("\tminute rate:    %s/second", formatter.format(totalCompactionsCompletedMetrics.getOneMinuteRate())));
+        toPrint.append(LINE_SEPARATOR);
+        toPrint.append(String.format("\t5 minute rate:    %s/second", formatter.format(totalCompactionsCompletedMetrics.getFiveMinuteRate())));
+        toPrint.append(LINE_SEPARATOR);
+        toPrint.append(String.format("\t15 minute rate:    %s/second", formatter.format(totalCompactionsCompletedMetrics.getFifteenMinuteRate())));
+        toPrint.append(LINE_SEPARATOR);
+        toPrint.append(String.format("\tMean rate:    %s/second", formatter.format(totalCompactionsCompletedMetrics.getMeanRate())));
+        toPrint.append(LINE_SEPARATOR);
+
+        return toPrint.toString();
+    }
+
+    private static String compactionThroughPutStats(NodeProbe probe)
+    {
+        Config conf = DatabaseDescriptor.loadConfig();
```

Comment:
Nodetool should work against any node; we shouldn't import DD at all and instead get this information from the probe.

Reply:
NodeProbe.getCompactionThroughputMebibytesAsDouble too reads from DatabaseDescriptor(via StorageService bean). Not sure what should be the correct source of data here.

Reply:
Yes, but over JMX from the node connected, not the local value, so it will be correct.

Reply:
So issue here is that I have to get hold of both configured value and actual value. I was under improession that NodeProbe.getCompactionThroughputMebibytesAsDouble would give me actual value.

Reply:
There isn't a running actual throughtput value available, you'll need to expose compactionRateLimiter.getRate in CompactionManager to JMX.